src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -0,0 +1,179 @@
+import org.elasticsearch.indices.breaker.CircuitBreakerService;
+import org.elasticsearch.indices.breaker.NoneCircuitBreakerService;
+import org.elasticsearch.indices.query.IndicesQueriesModule;
+import org.elasticsearch.script.ScriptModule;
+import org.elasticsearch.test.ElasticsearchTestCase;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.threadpool.ThreadPoolModule;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.io.IOException;
+
+@Ignore
+public abstract class BaseQueryTestCase<QB extends BaseQueryBuilder & Streamable> extends ElasticsearchTestCase {
+
+    private static Injector injector;
+    private static IndexQueryParserService queryParserService;
+    private static Index index;
+    protected QueryParseContext context;
+    protected XContentParser parser;
+
+    protected QB testQuery;
+
+    /**
+     * Setup for the whole base test class.
+     * @throws IOException
+     */
+    @BeforeClass
+    public static void init() throws IOException {
+        Settings settings = ImmutableSettings.settingsBuilder()
+                .put("name", BaseQueryTestCase.class.toString())
+                .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
+                .build();
+
+        index = new Index("test");
+        injector = new ModulesBuilder().add(
+                new EnvironmentModule(new Environment(settings)),
+                new SettingsModule(settings),
+                new ThreadPoolModule(settings),
+                new IndicesQueriesModule(),
+                new ScriptModule(settings),
+                new IndexSettingsModule(index, settings),
+                new IndexCacheModule(settings),
+                new AnalysisModule(settings),
+                new SimilarityModule(settings),
+                new IndexNameModule(index),
+                new IndexQueryParserModule(settings),
+                new FunctionScoreModule(),
+                new AbstractModule() {
+                    @Override
+                    protected void configure() {
+                        bind(ClusterService.class).toProvider(Providers.of((ClusterService) null));
+                        bind(CircuitBreakerService.class).to(NoneCircuitBreakerService.class);
+                    }
+                }
+        ).createInjector();
+        queryParserService = injector.getInstance(IndexQueryParserService.class);
+    }
+
+    /**
+     * Setup for the individual tests, creating test query, context and XContentparser
+     */
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        context = new QueryParseContext(index, queryParserService);
+        testQuery = createTestQueryBuilder();
+        String contentString = testQuery.toString();
+        parser = XContentFactory.xContent(contentString).createParser(contentString);
+        context.reset(parser);
```

Comment:
same comment as before, even more valid now that we need the parser for a single test. Shall we pull out these couple lines to a method that we call on demand where we need it instead of before each test?